### DISCUSSION TITLE
Memory Game Update

### DISCUSSION THREAD

**Discussion by gabebott** [USU-ITLS-x265-23-summer/class-repo#213] [General]
Hello again! Week 2 of working on the "MemoryBlitz" memory game has been productive, and I'm satisfied with the progress made during this time.

Progress:

JavaScript Logic:
The core game logic for card flipping, matching pairs, and tracking game statistics has been successfully implemented. Players can now flip two cards at a time, and the game checks for matches. If a pair matches, the cards remain face-up; otherwise, they flip back after a short delay.

Win Condition:
The win condition logic has been integrated into the game. When all card pairs are matched, a congratulatory message is displayed to the player, indicating their victory. Additionally, a button to restart the game is presented, allowing players to play again.

Sound Effects:
Exciting sound effects have been added to enhance the interactivity and enjoyment of the game. Players now hear a satisfying "flip" sound when cards are turned over and a cheering sound when they successfully match a pair.

Task List Status:
With the core functionality completed, the task list for Week 3 is looking promising. However, some additional tasks remain to polish the game and add more features to enhance the user experience.

Task List for Week 3:

Timer Functionality:
Implement a timer that starts when the player begins the game and stops when all card pairs are matched. Display the elapsed time alongside other game statistics.

User Interface Improvements:
Refine the user interface, ensuring consistency in design and responsiveness across different devices and screen sizes.

Difficulty Levels:
Introduce difficulty levels, allowing players to choose from easy, medium, or hard modes. Higher difficulty levels can include more cards and shorter time limits, offering an extra challenge.

High Score Storage:
Incorporate a mechanism to store and display the player's best scores. This could be achieved using browser storage or a server-based database.

Mobile Optimization:
Optimize the game for mobile devices, ensuring smooth gameplay and a user-friendly experience on smartphones and tablets.
Accessibility Considerations:

Bug Fixes and Testing:
Conduct thorough testing to identify and resolve any bugs or issues in the game.
Heading into next week, I'm excited about the upcoming tasks and the opportunity to refine "MemoryBlitz" even further. The game is shaping up well, and I can't wait to see it in its final form.

**Comment by jakegros** [comment]
Gabe, this sounds very exciting and impressive. You have certainly accomplished a lot.
Let me know if you want someone to test it out. I'm more than willing to give it a go.

**Comment by Skye404** [comment]
I echo what jakegros said. You've done a lot. I'd test it out for you too.

**Comment by CJ3315** [comment]
This game sounds awesome! I would also love to try!

**Comment by beto0890** [comment]
Great progress!

**Comment by derekscholes** [comment]
Has this been published yet or is it only available in the githubrepo? Id love to take a look when I have time as well
